### PR TITLE
[docs] GCP Cloud SQL backend reference

### PR DESCRIPTION
**Pull request by greedy52**
related:
- #41392

Preview:
https://docs-f4gsgx3mu-goteleport.vercel.app/docs/reference/backends/#google-cloud-iam-authentication

### CODE REVIEW

File: docs/pages/reference/backends.mdx [hunk 1]
```diff
@@ -391,6 +391,62 @@ credentials, or [managed
...
+flag `cloudsql.iam_authentication` and flag `cloudsql.logical_decoding` and are
+set to `on` for the Cloud SQL instance. The database user must also have the
+`REPLICATION` role attribute for using the logical decoding features. See [set
+up logical replication and
+decoding](https://cloud.google.com/sql/docs/postgres/replication/configure-logical-replication)
+for more details.
+
+In order for Teleport to use [Cloud SQL Go
+Connector](https://cloud.google.com/sql/docs/mysql/connect-connectors) with IAM
+authentication, the service account of the target databse user must have "Cloud
```
Comment (r0mant) [inline docs/pages/reference/backends.mdx:433]:
```suggestion
authentication, the service account of the target database user must have "Cloud
```

File: docs/pages/reference/backends.mdx [hunk 2]
```diff
@@ -391,6 +391,62 @@ credentials, or [managed
...
+
+    # GCP connection name is in format of <project>:<location>:<instance>.
+    gcp_connection_name: <Var name="project:location:instance"/>
+
+    # Type of IP for GCP connections. Valid options are:
+    # - "" (default to "public")
+    # - "public"
+    # - "private"
+    # - "psc" (for Private Service Connect)
+    gcp_ip_type: <Var name="public"/>
```
Comment (r0mant) [inline docs/pages/reference/backends.mdx:414]: Is this something users would need to set? How would they determine which one to pick?

**Review by r0mant — APPROVED** [review]

File: docs/pages/reference/backends.mdx (hunk 2 above)
Comment (greedy52) [reply docs/pages/reference/backends.mdx:414]: The owner of the Cloud SQL instance configures the IP mode when creating the database (or can modify it after)
Screenshot 2024-06-05 at 11 57 33 AM


So here `gcp_ip_type` depends on what IPs DB has and which is accessible to Auth.

File: docs/pages/reference/backends.mdx [hunk 3]
```diff
@@ -391,6 +391,62 @@ credentials, or [managed
...
+Authentication](https://cloud.google.com/sql/docs/postgres/iam-authentication)
+to connect to an GCP Cloud SQL for PostgreSQL without having to manage any
+secrets:
+```yaml
+teleport:
+  storage:
+    type: postgresql
+    auth_mode: gcp-cloudsql
+
+    # GCP connection name is in format of <project>:<location>:<instance>.
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:406]:
```suggestion
    # GCP connection name has the format <project>:<location>:<instance>.
```

File: docs/pages/reference/backends.mdx [hunk 4]
```diff
@@ -391,6 +391,62 @@ credentials, or [managed
...
+    gcp_connection_name: <Var name="project:location:instance"/>
+
+    # Type of IP for GCP connections. Valid options are:
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:409]: I would make it more explicit that this is the IP type for the user's Cloud SQL instance

File: docs/pages/reference/backends.mdx [hunk 5]
```diff
@@ -391,6 +391,62 @@ credentials, or [managed
...
+    # Leave host and port empty as they are not required.
+    conn_string: postgresql://<Var name="service-account@project.iam"/>@/teleport_backend
+
+    audit_events_uri:
+      - postgresql://<Var name="service-account@project.iam"/>@/teleport_audit#auth_mode=gcp-cloudsql&gcp_connection_name=<Var name="project:location:instance"/>&gcp_ip_type=<Var name="public"/>
+```
+
+To enable IAM authentication and logical replication for Cloud SQL, make sure
+flag `cloudsql.iam_authentication` and flag `cloudsql.logical_decoding` and are
+set to `on` for the Cloud SQL instance. The database user must also have the
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:425]:
```suggestion
To enable IAM authentication and logical replication for Cloud SQL, make sure the
flags `cloudsql.iam_authentication` and `cloudsql.logical_decoding` are
set to `on` for the Cloud SQL instance. The database user must also have the
```
Minor readability tweak

File: docs/pages/reference/backends.mdx [hunk 6]
```diff
@@ -391,6 +391,62 @@ credentials, or [managed
...
+`REPLICATION` role attribute for using the logical decoding features. See [set
+up logical replication and
+decoding](https://cloud.google.com/sql/docs/postgres/replication/configure-logical-replication)
+for more details.
+
+In order for Teleport to use [Cloud SQL Go
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:431]:
```suggestion
In order for Teleport to use the [Cloud SQL Go
```
From the linked docs, it looks like this requires an article

File: docs/pages/reference/backends.mdx [hunk 7]
```diff
@@ -391,6 +391,62 @@ credentials, or [managed
...
+Connector](https://cloud.google.com/sql/docs/mysql/connect-connectors) with IAM
+authentication, the service account of the target databse user must have "Cloud
+SQL Client"/`roles/cloudsql.client` and "Cloud SQL Instance
+User"/`roles/cloudsql.instanceUser` roles assigned to the service account.
+
+Teleport will make use of the credentials specified through
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:437]:
```suggestion
Teleport will make use of the credentials specified through the
```

**Review by ptgott — APPROVED** [review]

**Pull request merged**